app/src/test/java/com/mk/tasky/authentication/domain/usecase/ValidateEmailUseCaseTest.kt

```diff
@@ -0,0 +1,49 @@
+package com.mk.tasky.authentication.domain.usecase
+
+import com.google.common.truth.Truth
+import com.mk.tasky.authentication.domain.utils.EmailMatcher
+import org.junit.Before
+import org.junit.Test
+
+internal class ValidateEmailUseCaseTest {
+    private lateinit var validateEmail: ValidateEmailUseCase
+    private lateinit var emailMatcher: EmailMatcher
+
+    @Before
+    fun setUp() {
+        emailMatcher = object : EmailMatcher {
+            override fun matches(email: String): Boolean {
+                return email.contains("@") && email.contains(".")
+            }
+        }
+        validateEmail = ValidateEmailUseCase(emailMatcher)
+    }
+
+    @Test
+    fun `Email with no @ or dot, return false`() {
+        val email = "asd"
+        val result = validateEmail(email)
+        Truth.assertThat(result).isFalse()
+    }
+
+    @Test
+    fun `Email with no @, return false`() {
+        val email = "asd.com"
+        val result = validateEmail(email)
+        Truth.assertThat(result).isFalse()
+    }
+
+    @Test
+    fun `Email with no dot, return false`() {
+        val email = "asd@asd"
+        val result = validateEmail(email)
+        Truth.assertThat(result).isFalse()
+    }
+
+    @Test
+    fun `Valid email, return true`() {
+        val email = "asd@asd.com"
+        val result = validateEmail(email)
+        Truth.assertThat(result).isTrue()
+    }
+}
```

Comment:
Maybe I didn't make this clear enough in WhatsApp, but this test class does pretty much nothing. All it tests is your custom implementation of `EmailMatcher` in this test class, but tests should of course test real behavior of your app. 
As long as all your email matcher does is match an email (and no further validation), there's no point in testing this at least with a local unit test. Hope it's clear, if not please ask :)